src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -72,4 +74,18 @@ public static boolean matchesKeys(ResourceLocation loc, RegistryKey<?>... keys)
 				return true;
 		return false;
 	}
+
+	/**
+	 * <p>Registers a {@link IDispenseItemBehavior} that will perform the new behavior if the condition is met
+	 * and the behavior that was already in the registry if not.</p>
+	 * This works even if multiple mods add new behavior to the same item.
+	 *
+	 * @author abigailfails
+	 * */
+	public static void registerConditionalDispenseBehavior(Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior newBehavior) {
+		IDispenseItemBehavior oldBehavior = DispenserBlock.DISPENSE_BEHAVIOR_REGISTRY.get(item);
```

Comment:
This local variable is redundant.

Reply:
No it's not, because that needs to be stored when the method is called - if `DispenserBlock.DISPENSE_BEHAVIOR_REGISTRY.get(item)` is invoked inside of `dispense`, it'll just return the new behavior.